x-pack/plugins/security_solution/public/flyout/document_details/right/navigation.tsx

```diff
@@ -35,16 +49,57 @@ export const PanelNavigation: FC<PanelNavigationProps> = memo(({ flyoutIsExpanda
         scopeId,
       },
     });
+
+    storage.set(FLYOUT_STORAGE_KEYS.LEFT_PANEL_EXPANDED, {
+      ...localStorageLeftPanelExpanded,
+      [flyout]: true,
+    });
+
+    // we only want to keep track of the left side open while the user is still using the flyout.
+    // that way when choosing another alert while the flyout is open, its state isn't going to reset.
+    // we remove the value from local storage when flyout is closed.
+    onClose$.pipe(take(1)).subscribe(() => clearLocalStorage());
+
     telemetry.reportDetailsFlyoutOpened({
       location: scopeId,
       panel: 'left',
     });
-  }, [eventId, openLeftPanel, indexName, scopeId, telemetry]);
+  }, [
+    openLeftPanel,
+    eventId,
+    indexName,
+    scopeId,
+    storage,
+    localStorageLeftPanelExpanded,
+    flyout,
+    onClose$,
+    telemetry,
+    clearLocalStorage,
+  ]);
+
+  // automatically open left panel if it was saved in local storage
+  if (
+    storage.get(FLYOUT_STORAGE_KEYS.LEFT_PANEL_EXPANDED) &&
+    storage.get(FLYOUT_STORAGE_KEYS.LEFT_PANEL_EXPANDED)[flyout]
+  ) {
+    expandDetails();
+  }
+
+  // we remove the value from local storage when the component is unmounted
+  // this takes care of the case when the user navigates away from the page while the flyout is open or for example closing the browser window
+  useEffect(
```

Comment:
Don't know if it's caused by this `useEffect`, but in my local instance, the left panel persistence only work once 

https://github.com/elastic/kibana/assets/18648970/b5085a7d-a8ac-4512-9a54-0759bb375e18

Reply:
yup you are correct. This is some code I added late in the development and forgot to test again. The reason I added that logic is because I want to clear local storage when users navigate to other pages, but I just realized that this hook is called every time the flyout is unmounted, which happens when we open the flyout for another alert.
This is basically useless, I need to move this logic somewhere else